chain/client/src/client.rs

```diff
@@ -236,6 +293,15 @@ impl Handler<NetworkClientMessages> for ClientActor {
                 }
                 NetworkClientResponses::NoResponse
             }
+            NetworkClientMessages::AnnounceAccount(announce_account) => {
+                if self.check_signature_account_announce(&announce_account) {
+                    let _ =
+                        self.network_actor.send(NetworkRequests::AnnounceAccount(announce_account));
+                    NetworkClientResponses::NoResponse
+                } else {
+                    NetworkClientResponses::Ban { ban_reason: ReasonForBan::InvalidSignature }
```

Comment:
Could also be invalid hash. Shall we return Result instead in`check_signature_account_announce`, and propagate the error here?

Reply:
Addressed on https://github.com/nearprotocol/nearcore/pull/1058/commits/50be2ac4ec71370b84ecc459127ec6d3d19bec00.